File: Python/7py 2/square_digits.py
"""
|--------------------------------------------------------------------------
| Square Every Digit
|--------------------------------------------------------------------------
|
"""

def square_digits(num):
   arr = list(str(num))
   res = ''
   for i in range(len(arr)):
      temp = int(arr[i]) ** 2
      res += str(temp)
   return int(res)


def square_digits(num):
   return int(''.join(str(int(d)**2) for d in str(num)))


def square_digits(num):
   return int(''.join(map(lambda x: str(int(x)**2), str(num))))


print(square_digits(9119), 811181)
print(square_digits(0), 0)

File: PHP/7php 3/square_digits.php
<?php
/*
|--------------------------------------------------------------------------
| Square Every Digit
|--------------------------------------------------------------------------
|
*/

function square_digits($num){
   $number = str_split($num);
   $sum = '';
   foreach ($number as $i) {
      $sum .= pow($i, 2); 
   }
   return $sum;
}


function square_digits($num){
   return implode('',array_map(
      function($n){
         return $n*$n;
      },str_split($num))
   );
}


function square_digits($num){
   $n = strval($num);
   $w = '';
   for ($i=0;$i<strlen($n);$i++) {
      $w .= strval(intval($n[$i]**2));
   }
   return intval($w);
}


print_r(square_digits(3212) . ' ' . 9414 . '
');
print_r(square_digits(2112) . ' ' . 4114 . '
');
print_r(square_digits(0) . ' ' . 0 . '
');